src/Compilers/CSharp/Portable/Binder/BlockBinder.cs

```diff
@@ -10,27 +10,27 @@ namespace Microsoft.CodeAnalysis.CSharp
 {
     internal sealed class BlockBinder : LocalScopeBinder
     {
-        private readonly SyntaxList<StatementSyntax> _statements;
+        private readonly BlockSyntax _block;
 
-        public BlockBinder(Binder enclosing, SyntaxList<StatementSyntax> statements)
-            : this(enclosing, statements, enclosing.Flags)
+        public BlockBinder(Binder enclosing, BlockSyntax block)
+            : this(enclosing, block, enclosing.Flags)
         {
         }
 
-        public BlockBinder(Binder enclosing, SyntaxList<StatementSyntax> statements, BinderFlags additionalFlags)
+        public BlockBinder(Binder enclosing, BlockSyntax block, BinderFlags additionalFlags)
             : base(enclosing, enclosing.Flags | additionalFlags)
         {
-            _statements = statements;
+            _block = block;
```

Comment:
> Do you think the assert will add value here?

Possibly.  Maybe I just misunderstand what the intended usage for `Debug.Assert` is in the compiler.  Looking at the various usages it's hard for me to come up with a rule of how / when we use it.